chain/network/src/test_loop.rs

```diff
@@ -1,4 +1,307 @@
+                    } else {
+                        return PeerManagerMessageResponse::NetworkResponses(
+                            NetworkResponses::NoResponse,
+                        );
+                    }
+                }
+                // If no handler was able to handle the request, panic.
+                if let Some(request) = request {
+                    panic!("Unhandled request: {:?}", request);
+                }
+            }
+            msg => panic!("Unexpected message: {:?}", msg),
+        };
+        PeerManagerMessageResponse::NetworkResponses(NetworkResponses::NoResponse)
+    }
+}
+
+fn network_message_to_client_handler(
+    my_account_id: &AccountId,
+    client_senders: HashMap<AccountId, ClientSenderForTestLoopNetwork>,
+) -> NetworkRequestHandler {
+    let my_account_id = my_account_id.clone();
+    Arc::new(move |request| match request {
+        NetworkRequests::Block { block } => {
+            for (account_id, sender) in client_senders.iter() {
+                if account_id != &my_account_id {
+                    let _ = sender.send_async(BlockResponse {
+                        block: block.clone(),
+                        peer_id: PeerId::random(),
+                        was_requested: false,
+                    });
+                }
+            }
+            None
+        }
+        NetworkRequests::Approval { approval_message } => {
+            assert!(
+                approval_message.target != my_account_id,
+                "Sending message to self not supported."
+            );
+            let sender = client_senders.get(&approval_message.target).unwrap();
+            let _ = sender.send_async(BlockApproval(approval_message.approval, PeerId::random()));
+            None
+        }
+        NetworkRequests::ForwardTx(account, transaction) => {
+            assert!(account != my_account_id, "Sending message to self not supported.");
+            let sender = client_senders.get(&account).unwrap();
+            let _ = sender.send_async(ProcessTxRequest {
+                transaction,
+                is_forwarded: true,
+                check_only: false,
+            });
+            None
+        }
+        NetworkRequests::ChunkEndorsement(target, endorsement) => {
+            assert!(target != my_account_id, "Sending message to self not supported.");
+            let sender = client_senders.get(&target).unwrap();
+            let _ = sender.send_async(ChunkEndorsementMessage(endorsement));
+            None
+        }
+        _ => Some(request),
+    })
+}
+
+fn network_message_to_partial_witness_handler(
+    my_account_id: &AccountId,
+    partial_witness_senders: HashMap<AccountId, PartialWitnessSenderForNetwork>,
+) -> NetworkRequestHandler {
+    let my_account_id = my_account_id.clone();
+    Arc::new(move |request| match request {
+        NetworkRequests::ChunkStateWitnessAck(target, witness_ack) => {
+            // TODO(testloop): Convert account check to a panic.
```

Comment:
@tayfunelmas, once we have your changes in for the partial witness distribution, we can change this to an assert like the rest of the places.